core/embed/rust/src/micropython/logging.rs

```diff
@@ -0,0 +1,94 @@
+use core::{ffi::CStr, str::from_utf8};
+
+use crate::{
+    error::Error,
+    micropython::{
+        buffer::StrBuffer, ffi, map::Map, module::Module, obj::Obj, print::print, qstr::Qstr, util,
+    },
+    strutil,
+    trezorhal::time::ticks_ms,
+};
+
+fn _log(level: &str, args: &[Obj], kwargs: &Map) -> Result<Obj, Error> {
+    if args.len() < 2 {
+        return Err(Error::TypeError);
+    }
+    {
+        let millis = ticks_ms();
+        let seconds = millis / 1000;
+        let mut millis_str = [b'0'; 3];
+        let len = unwrap!(strutil::format_i64((millis % 1000).into(), &mut millis_str)).len();
+        millis_str.rotate_left(len);
+        let log_prefix = uformat!(len: 128, "{}.{} \x1b[35m{}\x1b[0m \x1b[{}\x1b[0m ",
+            seconds, unwrap!(from_utf8(&millis_str)), StrBuffer::try_from(args[0])?.as_ref(), level,
+        );
+        print(&log_prefix);
+    }
+
+    if let Ok(iface_obj) = kwargs.get(Qstr::MP_QSTR_iface) {
+        if iface_obj != Obj::const_none() {
+            // SAFETY: every MicroPython object has a type with a valid name string
+            let iface_cstr = unsafe {
+                let ptr = ffi::mp_obj_get_type_str(iface_obj.as_ptr());
```

Comment:
Can't we do something like `iface_obj.type_().name()` instead? If yes then let's get rid of the unsafe block.

Reply:
Thanks! 
Fixed in acbd7e70874ab8df8aeec213e2f83bc22c4d0595.